tc/core/polyhedral/cuda/mapped_scop.cc

```diff
@@ -267,11 +267,11 @@ bool MappedScop::detectReductions(detail::ScheduleTree* tree) {
   updates.foreach_set([&updateIds](isl::set set) {
     updateIds.emplace_back(set.get_tuple_id());
   });
-  // The reduction member needs to appear right underneath
-  // the coincident members.
-  auto reductionDim = nCoincident;
-  auto member = band->mupa_.get_union_pw_aff(reductionDim);
-  if (!isReductionMember(member, updates, scop())) {
+  // The coincident members, together with the prefix schedule,
+  // need to determine a single reduction.
+  auto prefix = prefixScheduleMupa(schedule(), tree);
+  prefix = prefix.range_product(band->memberRange(0, nCoincident));
```

Comment:
This still assumes coincident band members appear before the reduction member, so this sort of contradicts the claim in the commit message of 90820bf66eaa90820a51f22c0f0b635938ef39d1 "It is not sufficient because it doesn't force the schedule dimensions not involving the reduction dimensions to appear outside of those band members."  This does not seem to force or check that property either.

Reply:
> This still assumes coincident band members appear before the reduction member,

It only assumes that within a band the coincident members appear
before the non-coincident members.  This is a property of
individual bands in the output of the isl scheduler and
is not directly related to reductions.

> so this sort of contradicts the claim in the commit message of 90820bf66eaa908
20a51f22c0f0b635938ef39d1 "It is not sufficient because it doesn't force the sch
edule dimensions not involving the reduction dimensions to appear outside of tho
se band members."  This does not seem to force or check that property either.

The check happens on the next line:
`if (!isSingleReductionWithin(updates, prefix, scop()))`